pkg/crc/config/viper_config.go

```diff
@@ -207,12 +179,17 @@ func (c *Config) Unset(key string) (string, error) {
 		return "", fmt.Errorf(configPropDoesntExistMsg, key)
 	}
 
-	delete(c.changedConfigs, key)
-	if err := syncViperState(c.configFile, c.changedConfigs, c.globalViper); err != nil {
-		return "", fmt.Errorf("Error unsetting configuration property '%s': %v", key, err)
+	settings := c.globalViper.AllSettings()
+	delete(settings, key)
+	bin, err := json.Marshal(settings)
+	if err != nil {
+		return "", err
+	}
+	if err = c.globalViper.ReadConfig(bytes.NewReader(bin)); err != nil {
+		return "", err
 	}
```

Comment:
might be clearer to just keep `syncViperState` a bit longer?

Reply:
I added more details in the commit message and split the commit in two: one with viper, one with the cast.